tests/src/test/scala/cats/tests/FoldableTests.scala

```diff
@@ -214,17 +209,23 @@ class FoldableTestsAdditional extends CatsSuite {
   }
 
   test(".foldLeftM short-circuiting optimality") {
-    // test that no more elements are evaluated than absolutely necessary
+    // test that no more than 1 elements are evaluated than absolutely necessary
```

Comment:
Should we look if we can solve this additional evaluation with @TomasMikula's [suggestion](https://github.com/typelevel/cats/pull/1117/files#r123775020)?